libclamav/others.h

```diff
@@ -1032,4 +1032,99 @@ cl_error_t cli_get_filepath_from_filedesc(int desc, char **filepath);
  */
 cl_error_t cli_realpath(const char *file_name, char **real_filename);
 
+
+
+
+#ifndef DECLARE_POINTER
+#define DECLARE_POINTER(type, name) type name = NULL
+#endif
+
+#ifndef STRDUP
+#define STRDUP(buf, var, ...)                  \
+    do {                                       \
+        var = strdup(buf);                     \
+        if (NULL == var) {                     \
+            do {                               \
+                __VA_ARGS__;                   \
```

Comment:
Wow this VA_ARGS thing to print error messages and set return status codes is pretty wild. I don't think I've seen that before.  😲  I think I like it.